supervision/detection/core.py

```diff
@@ -1066,6 +1068,33 @@ def __setitem__(self, key: str, value: Union[np.ndarray, List]):
 
         self.data[key] = value
 
+    def _set_at_index(self, index: int, other: Detections):
```

Comment:
Wouldn't placing this code as part of the __setitem__ method makes more sense? The flow below feels quite natural to me. 

```python
detections_1 = sv.Detections(...)
detections_2 = sv.Detections(...)
detections_1[0] = detections_2[0]
```

Reply:
`__setitam__`

```
detections_2[0]
detections_2["class_name"]
```

`__getitam__`

```
detections_2[0]
detections_2[1:3]
detections_2[[1, 2, 3]]
detections_2[[False, True, False]]
detections_2["class_name"]
```

Reply:
_set_at_index was not required. I removed it entirely, and did not add any logic to __setitem__.